libbpf-rs/src/map.rs

```diff
@@ -731,3 +734,117 @@ impl<'a> Iterator for MapKeyIter<'a> {
         }
     }
 }
+/// Represent a wrapper `bpf_map_info` struct pointer
+/// Provide ability to retrive the details of a certain map with its fd only
+#[derive(Debug)]
+pub struct MapInfo {
+    info: bpf_map_info,
+}
+
+impl MapInfo {
+    /// Create a MapInfo struct from a fd.
+    pub fn new(fd: i32) -> Result<Self> {
```

Comment:
Let's stop using "raw" file descriptors and use `std::os::fd::BorrowedFd` or `OwnedFd` here as applicable.

Reply:
The fd may be not obtained from an bpf_map* struct, and even can be provided by the user. So maybe it's not a good idea to use wrapped file descriptors?

Reply:
> The fd may be not obtained from an bpf_map* struct, and even can be provided by the user.

Yeah, but that's why I'd say it's important to make the contract clear whether it's consuming the file descriptor or not.

Reply:
> > The fd may be not obtained from an bpf_map* struct, and even can be provided by the user.
> 
> Yeah, but that's why I'd say it's important to make the contract clear whether it's consuming the file descriptor or not.

I'm using `BorrowedFd` now.